board.cpp

```diff
@@ -0,0 +1,205 @@
+                    break;
+                }
+            }
+        }
+    }
+    if(mineInPos == true) std::cout << "  M";
+    else if(isPositionEnabled==false) std::cout << "   ";
+    else if(isPositionEnabled == true) std::cout << "  O";
+}
+
+void printBoardAuxiliars::printRow(Board const& board, unsigned int y){
+    int spaceLength = (3 - std::to_string(y).length());
+    if (y == 0){ //si es la primera fila, imprimimos solo espacio
+            std::cout << "   ";
+        }
+    else{ //caso contrario, imprimimos el numero de la fila
+        std::cout << y;
+        for(int space = 0; space < spaceLength; space++){
+            std::cout << ' ';
+        };
+    }
+}
+
+void printBoardAuxiliars::printColumnInRow(Board const& board, unsigned int x, unsigned int y, int perspective){
+    int spaceLength = (3 - std::to_string(x).length());
+    if (y == 0){ //si es la primera fila, imprimimos los numeros de las columnas
+        for(int space = 0; space < spaceLength; space++){
+            std::cout << ' ';
+        };
+        std::cout << std::to_string(x);
+    }
+    else{ //en caso contrario, imprimimos el estado de la posicion 
+        printBoardAuxiliars::showPositionStatus(board, x, y, perspective);
+    }
+    return;
+}
+
+//print board to console
+//may be depreciated if an UI is implemented in the future
+void printBoard(Board const& board, int perspective){
+    for (int y = 0; y <= board.height; y++){ //para cada fila
+        printBoardAuxiliars::printRow(board, y);
+        for (int x = 1; x <= board.width; x++){ // seguido del numero de fila, imprimimos todas las posiciones de la fila
+            printBoardAuxiliars::printColumnInRow(board, x, y, perspective);
+        }
+        std::cout << std::endl;
+    }
+}
+
+void disablePosition(Board &board, Position disabledPosition){
+    board.disabledPositions.push_back(disabledPosition);
+}
+
+bool removeMine(Board &board, Mine mine){
+    for(std::vector<Mine>::iterator it = board.placedMines.begin(); it != board.placedMines.end(); it++){
+        if (*it == mine){
+            board.placedMines.erase(it);
+            board.players[mine.owner].mineCount--;
+            std::cout << "Player " << mine.owner + 1 << " mines remaining: " << board.players[mine.owner].mineCount << std::endl;
+            disablePosition(board, mine);
+            return true;
+        }
+    }
+    return false; //just in case I need to check if a mine was properly erased or not
+}
+
+//function to be called at the end of each turn
+void disableTilesUsed(Board &board){ 
+    for(std::vector<Mine>::iterator it = board.placedMines.begin(); it != board.placedMines.end(); it++){
+        disablePosition(board, *it);
+    }
+}
+
+//when only one player has mines remaining, they win the game
+//if no players have mines, game is a draw
+//TODO: refactor to support more than 2 players
+int getWinningPlayer(Board const& board){
+    int winner = -1;
+    if(board.players[0].mineCount <= 0){

```

Comment:
#must do bounds checking before continuing with this function (that is, make sure that the length of the `players` vector is at least `2`)